klippy/extras/resonance_tester.py

```diff
@@ -32,49 +32,54 @@ def get_supported_axes(self):
         return ['x', 'y']
     def get_start_test_points(self):
         return self.probe_points
-    def prepare_test(self, toolhead, gcmd):
+    def prepare_test(self, gcmd):
         self.freq_start = gcmd.get_float("FREQ_START", self.min_freq, minval=1.)
         self.freq_end = gcmd.get_float("FREQ_END", self.max_freq,
                                        minval=self.freq_start, maxval=200.)
         self.hz_per_sec = gcmd.get_float("HZ_PER_SEC", self.hz_per_sec,
                                          above=0., maxval=2.)
-        # Attempt to adjust maximum acceleration and acceleration to
-        # deceleration based on the maximum test frequency.
-        max_accel = self.freq_end * self.accel_per_hz
-        toolhead.cmd_SET_VELOCITY_LIMIT(self.gcode.create_gcode_command(
-            "SET_VELOCITY_LIMIT", "SET_VELOCITY_LIMIT",
-            {"ACCEL": max_accel, "ACCEL_TO_DECEL": max_accel}))
-    def run_test(self, toolhead, axis, gcmd):
+    def run_test(self, axis, gcmd):
+        toolhead = self.printer.lookup_object('toolhead')
         X, Y, Z, E = toolhead.get_position()
         if axis not in self.get_supported_axes():
             raise gcmd.error("Test axis '%s' is not supported", axis)
         vib_dir = (1, 0) if axis == 'x' else (0., 1.)
         sign = 1.
         freq = self.freq_start
+        # Override maximum acceleration and acceleration to
+        # deceleration based on the maximum test frequency
+        _, old_max_accel = toolhead.get_max_velocity()
+        max_accel = self.freq_end * self.accel_per_hz
+        toolhead.cmd_SET_VELOCITY_LIMIT(self.gcode.create_gcode_command(
+            "SET_VELOCITY_LIMIT", "SET_VELOCITY_LIMIT",
+            {"ACCEL": max_accel, "ACCEL_TO_DECEL": max_accel}))
```

Comment:
FWIW, I think it might be a little simpler to call `self.gcode.run_script_from_command('SET_VELOCITY_LIMIT ACCEL=%.3f ACCEL_TO_DECEL=%.3f' % (max_accel, max_accel))`

Reply:
Indeed, thanks!